SPR modifications to support api-v2 (Spack v1.x)

harshula opened the pull request:
DRAFT to track changes

harshula committed:
Make package directory names API v2.0 compliant

* Hyphen is replaced by underscore.

harshula committed:
spack repo migrate --fix

harshula committed:
access_mocsy, access3_share, cice5, mom5: fix namespace

harshula committed:
Support Spack v1.0 by depending on languages

harshula committed:
Support Spack v1.0 compilers as a package

harshula committed:
Make directory path API v2.0 compliant

harshula committed:
module llnl has been moved to spack.llnl

* https://github.com/spack/spack/ PR 50989

harshula committed:
Add __init__.py files and spack-repo-index.yaml

* __init__.py: review feedback from Peter Scheibel.
* spack-repo-index.yaml: suggested by Tommy Gatti.

CodeGat commented:
Hey @harshula, the error is because of the changed directory structure for `spack-packages` between `api-v1` and `api-v2`. Unfortunately I can't suggest changes directly because the bits I want to change are unmodified. But:

In https://github.com/ACCESS-NRI/spack-packages/blob/cc4ec27340bc179b1efade7c74938e49ea61d5b6/.github/workflows/ci.yml#L5, you need to change that to `- spack_repo/access/nri/packages/**`

In https://github.com/ACCESS-NRI/spack-packages/blob/cc4ec27340bc179b1efade7c74938e49ea61d5b6/.github/workflows/ci.yml#L51, change it to `path: ./spack_repo/access/nri/packages`

CodeGat commented:
Just noting that I've updated the upstream image in the cluster to `:rocky-v1.0-2025.11.000`, which contains the `intel-oneapi-compilers-classic@2021.10.0 target=x86_64` compiler.

CodeGat commented:
@harshula, once this run is done, feel free to up the `max-parallel` to `7`

CodeGat commented:
I've updated the image in the infrastructure to `:rocky-v1.1-2025.12.000-test`, please use that as the `container-image-version`. Furthermore, that image is only compatible with the new version of `build-ci` currently being tested, so it might be good to knock two birds with one stone.

CodeGat commented:
Hi @harshula, it was not - I'll recreate the image without the compilers and update the infra!

CodeGat commented:
There was a bug in `build-ci` which led to the wrong upstream being selected from the kubernetes cluster. This has now been fixed, and would explain the mismatch between the hashes found by the runner in the upstream (which actually came from a `0.22` upstream) and the hashes in the actual upstream needed (the `1.X` one).

CodeGat commented:
So you're saying...that's that? It works? :astonished:

CodeGat committed:
`build-ci v3`: Update Infra For `spack v1` (#353)

infra: Update to v3, get default spack-config-ref/spack-ref from build-ci workflow, use default container-image-version

harshula committed:
CI: use Spack v1.1

* .github/build-ci/data/standard_definitions.json:
  * Change syntax to suit toolchains
* Update spack.yaml templates to use Spack v1.x compiler toolchains syntax
* .github/workflows/ci.yml:
  * DEFAULT_SPACK_REF: 'releases/v1.1'
* Add a generic intel-oneapi-compilers test
* %intel@2021.10.0 can not compile the latest version of python
  * Require python 3.11.14.
* max-parallel increased from 3 to 5.

* PR 346:
  * Update paths to `api-v2` format
  * Replace `_` from directory name with `-` for package names
  * Swap filepath and template_value so package name comes first in matrix
  * Consolidate all refs to packages path under env.PACKAGES_ROOT_DIR

Co-authored-by: Tommy Gatti <tommy.gatti@anu.edu.au>

CodeGat reviewed (approved):
I'm going to pre-emptively approve this even though the checks haven't finished, that's how hopeful I am. Even though this branch will be the next default branch.
